static/js/Sheets/SheetsWithRefPage.jsx

```diff
@@ -0,0 +1,191 @@
+import SearchPage from "../SearchPage";
+import Sefaria from "../sefaria/sefaria";
+import {useEffect, useState} from "react";
+import {SearchTotal} from "../sefaria/searchTotal";
+import {Children as availableFilters} from "../lib/react";
+const SheetsWithRefPage = ({srefs, searchState, updateSearchState, updateAppliedFilter,
+                           updateAppliedOptionField, updateAppliedOptionSort, onResultClick,
+                           registerAvailableFilters}) => {
+    const [sheets, setSheets] = useState([]);
+    const [loading, setLoading] = useState(true);
+    const [origAvailableFilters, setOrigAvailableFilters] = useState([]);
+    const [refs, setRefs] = useState(srefs);
+    const cloneFilters = (availableFilters, resetDocCounts = true) => {
+        return availableFilters.map(availableFilter => {
+            let newAvailableFilter = availableFilter.clone();
+            if (resetDocCounts) newAvailableFilter.docCount = 0;
+            return newAvailableFilter;
+        })
+    }
+    const getDocCounts = (availableFilters) => {
+        return availableFilters.map(availableFilter => availableFilter.docCount).sort((a, b) => a - b);
+    }
+    const checkForRegisteringAvailableFilters = (availableFilters) => {
+        const newDocCounts = getDocCounts(availableFilters);
+        const currDocCounts = getDocCounts(searchState.availableFilters);
+        if (!newDocCounts.compare(currDocCounts)) {
+            availableFilters = availableFilters.sort((a, b) => b.docCount - a.docCount || a.title.localeCompare(b.title));
+            registerAvailableFilters('sheet', availableFilters, {}, [], ['collections', 'topics_en']);
+        }
+    }
+    const updateDocCounts = (newAvailableFilters, slugs) => {
+        newAvailableFilters.forEach((newFilter, i) => {
+            if (newFilter.aggKey in slugs) {
+                newAvailableFilters[i].docCount = slugs[newFilter.aggKey];
+            }
+        })
+    }
+    const getSheetSlugs = (type, sheet) => {
+        const items = type === 'topics_en' ? sheet.topics : sheet.collections;
+        return items.map(x => x.slug);
+    }
+    const applyFiltersToSheets = (sheets) => {
+        searchState.appliedFilters.forEach((appliedFilter, i) => {
```

Comment:
Yeah it works, unless I'm misunderstanding the question.  You can select multiple filters and it seems to filter properly

Reply:
For applied filters "Shabbat" and "Education", it will only display sheets that have BOTH of these filters on.  That seems to be the way the filters work for searching for sheets, so I copied that behavior

Reply:
Ah, you're right. This logic looks sound.